src/sagas/assignments.js

```diff
@@ -0,0 +1,71 @@
+import {all, call, put, select, takeEvery} from 'redux-saga/effects';
+
+import {
+  getCourses,
+  createClassroomAssignment,
+} from '../clients/googleClassroom';
+import {
+  createProjectSnapshot,
+} from '../clients/firebase';
+import {
+  assignmentCreated,
+  assignmentNotCreated,
+} from '../actions/assignments';
+import {
+  closeAssignmentCreator,
+  updateCourses,
+} from '../actions/ui';
+import {
+  getCurrentProject,
+} from '../selectors';
+import {generateTextPreview} from '../util/compileProject';
+import {createSnapshotUrl} from '../util/exportUrls';
+
+export function* openAssignmentCreator() {
+  const courses = yield call(getCourses);
+  yield put(updateCourses(courses));
+}
+
+export function* assignAssignment(
+  {payload: {selectedCourse, dateInput}}) {
+  yield* createAssignment(
+    selectedCourse,
+    dateInput,
+    'PUBLISHED',
```

Comment:
Should we only use the createassignment function and add the assignment type in when the action is dispatched?